vars/fhcapProviderConfigFor.groovy

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/groovy
+
+def call(providerId, providers, template = 'default') {
+    def providerCfg = null
+    String delimiter = '-'
+    List<String> tmplParts = template.split(delimiter) as List<String>
+    while(tmplParts.size() > 0) {
+        String tmplStr = tmplParts.join(delimiter)
+        providerCfg = providers[providerId].providerConfig[tmplStr]
+        if(providerCfg) {
+            break
+        }
+        tmplParts.pop()
+    }
+    (providers[providerId].providerConfig['default'] ?: [:]) + (providerCfg ?: [:])
+}
```

Comment:
This is currently only searching for at least one match starting with the template name and working backwards, but maybe a better solution is to go the other way and merge the result from each gradually building up an object.